src/ILLink.RoslynAnalyzer/DataFlow/LocalDataFlowVisitor.cs

```diff
@@ -221,5 +201,49 @@ public override TValue VisitConversion (IConversionOperation operation, LocalDat
 			var operandValue = Visit (operation.Operand, state);
 			return operation.OperatorMethod == null ? operandValue : TopValue;
 		}
+
+		public override TValue VisitObjectCreation (IObjectCreationOperation operation, LocalDataFlowState<TValue, TValueLattice> state)
+		{
+			if (operation.Constructor == null)
+				return TopValue;
+
+			// Passing null for instance value is a bit wrong - below we will translate it to TopValue right now.
+			// Technically this is an instance call and the instance is some valid value, we just don't know which
+			// but for example it does have a static type. For now this is OK since we don't need the information
+			// for anything yet.
+
+			// The return here is also technically problematic, the return value is an instance of a known type.
```

Comment:
nit: Would you mind moving this comment to `HandleMethodCall`? (Since this code is generic and doesn't know about DAM analysis, and also because that's where the issue would probably get fixed).